src/common_fits.jl

```diff
@@ -30,14 +23,73 @@ immutable FitResult
     model_str::String
 end
 
-fit_dict1(p) = Dict("a"=>p[1], "T"=>p[2], "f"=>p[3], "ϕ"=>p[4], "b"=>p[5])
+"""
+   `generic_fit(xpts, ypts, model, initial_guess, fit_params, model_string; yvars=[])`
+
+Helper function to return a fit to a model.
 
-fit_dict2(p) = Dict("a₁"=>p[1], "T₁"=>p[2], "f₁"=>p[3], "ϕ₁"=>p[4],
-                    "a₂"=>p[5], "T₂"=>p[6], "f₂"=>p[7], "ϕ₂"=>p[8],
-                    "b"=>p[9])
+# Arguments
+* xpts: Independent fit variable.
+* ypts: Dependent fit variable.
+* model: Fit function(x, p) of data x and parameter vector p.
+* initial_guess: Initial guess of fit parameters.
+* fit_params::function(Array{Float64} => Dict{String, Float64}): Result dict of fit parameters.
+* model_string::String: String indicating fit function.
+* yvars: Variance of y points. Defaults to empty vector.
 
+# Returns
+* A FitResult object for the specified fit.
 """
-        fit_ramsey(xpts, ypts, yvars=[])
+function generic_fit(xpts, ypts, model, initial_guess, fit_params, model_string::String; yvars=[])
+	@assert length(xpts) == length(ypts) "X and Y data length must match."
+	if isempty(yvars)
+			result = curve_fit(model, xpts, ypts, initial_guess)
+			errors = estimate_errors(result)
+			sq_error = sum(((model(xpts, result.param) - ypts).^2))
+	else
+			@assert length(ypts) == length(yvars) "Y data and Y variance lengths must match."
+			result = curve_fit(model, xpts, ypts, 1./sqrt(yvars), initial_guess)
+			errors = estimate_errors(result)
+			sq_error = sum(((model(xpts, result.param) - ypts).^2)./yvars)
+	end
+	# Compute badness of fit:
+	# Under the null hypothesis (that the model is valid and that the observations
+	# do indeed have Gaussian statistics), the mean squared error is χ² distributed
+	# with `dof` degrees of freedom. We can quantify badness-of-fit in terms of how
+	# far the observed MSE is from the expected value, in units of σ = 2dof (the expected
+	# standard deviation for the χ² distribution)
+	dof = length(xpts)-length(initial_guess)
+	Nσ = sq_error/sqrt(2dof) - dof/sqrt(2dof)
+	return FitResult( fit_params(result.param),
+										sq_error,
+										Nσ,
+										fit_params(errors),
+										xpts->model(xpts,result.param),
+										model_string)
+end
+
+""" `fit_t1(xpts, ypts, yvars=[])`
+Fit to exponential T₁ decay y = a exp(-t/T₁) + b.
+"""
+function fit_t1(xpts, ypts, yvars=[])
+	T1_fit_dict(p) = Dict("a" => p[1], "T₁" => p[2], "b" => p[3])
```

Comment:
I would dispense with subscripts when unnecessary. I would just use `T` for simplicity